api_tests/tests/ether_halight.ts

```diff
@@ -36,3 +36,40 @@ it(
         await bob.assertBalances();
     })
 );
+
+it(
+    "han-ethereum-ether-halight-lightning-bitcoin-alice-announces-with-wrong-peer-id",
+    twoActorTest(async ({ alice, bob }) => {
+        const bodies = (await SwapFactory.newSwap(alice, bob))
+            .hanEthereumEtherHalightLightningBitcoin;
+
+        // Simulate that Bob is awaiting a swap from a different peer-id than Alice node's peer-id.
+        bodies.bob.peer.peer_id =
+            "QmYyQSo1c1Ym7orWxLYvCrM2EmxFTANf8wXmmE7DWjhx5N";
+        await bob.createSwap(bodies.bob);
+        await sleep(500);
+
+        await alice.createSwap(bodies.alice);
+
+        const config = {
+            maxTimeoutSecs: 1,
+            tryIntervalSecs: 0.3,
+        };
+
+        // Due to the asynchronous nature of the announce protocol (and the fact that there is not error handling (yet)),
+        // this test only tests that the "init" action does not become available.
+        //
+        // Scenario details:
+        // 0) Alice and Bob agree on the swap params (negotiation); Alice gives Bob a peer-id that is not her node's peer-id.
+        // 1) Bob posts swap, retrieves local swap-ID
+        // 2) Alice posts swap, retrieves local swap-ID
+        //      2.1) Internally the announce protocol runs into an Error because Alice' peer-id does not match the one she gave Bob.
+        //              This Error, is however not visible to the outside.
+        //              The application is responsible for removing swaps that do not move forward after a given time.
+        const aliceResponsePromise = alice.init(config);
+        return expect(aliceResponsePromise).rejects.toThrowError();
```

Comment:
You can only return once from a function :)

```suggestion
        await expect(aliceResponsePromise).rejects.toThrowError();
```